torch/_subclasses/fake_tensor.py

```diff
@@ -1749,6 +1706,24 @@ def _make_cache_entry(
         _BypassDispatchCache if the output tensor has characteristics that
         prevent caching it.
         """
+        from torch._higher_order_ops.utils import registered_hop_fake_fns
+
+        # For hops, lets look at the output tensor to find any unbacked symints.
+        # If there are none, then we rely on the existing checks to validate
+        # caching.
+        if (
+            isinstance(func, torch._ops.HigherOrderOperator)
+            and func in registered_hop_fake_fns
+        ):
+            assert isinstance(output, tuple)
+            non_cacheable = any(
+                isinstance(o, (torch.Tensor, torch.SymInt))
+                and has_free_unbacked_symbols(o)
+                for o in output
+            )
+            if non_cacheable:
```

Comment:
we need to go through all of the existing bypass reasons and think through them with respect to `invoke_subgraph`.

I think we talked through most of them offline, but another one is - similar to `lift_fresh_copy` handling - do we still cache miss properly on `invoke_subgraph` if any of its inputs are real tensors?

Reply:
Fair enough. I will go over others in a meeting.

Why do we need to miss on lift fresh copy though? I think as long as the output is not affected, it should still be cacheable.

Reply:
Hmm I looked through the places where we raise a cache miss and I guess we did cover all of them.

and on lift_fresh- the original idea was that the cache needs to key on input metadata, and “metadata” for a tensor constant (the input to lift_fresh) is effectively the full data of the tensor. I think that isn’t really true though? The output faketensor wouldn’t depend on the input tensor data, unless there were data-dependent shapes in the output.

So tldr feel free to ignore this comment

Reply:
Yeah, in my case, the lift is used mostly for attrs inside the subgraph and they are used for some very silly computation and then consumed.